### PR TITLE
Update the WebIDL parser

### PR DESCRIPTION
**Pull request by nox**
It now supports Python 3.

### CODE REVIEW

**Comment by CYBAI** [comment]
Maybe this will fix #25955 👀?

**Comment by nox** [comment]
It updates WebIDL.py but it doesn't make the script crate work with Python 3. And mozjs_sys and style also don't build with Python 3.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -1665,6 +1654,8 @@ def __init__(self, descriptor, name, static, unforgeable):
             (maplikeOrSetlikeOrIterable and
              maplikeOrSetlikeOrIterable.isIterable() and
              maplikeOrSetlikeOrIterable.isValueIterator())):
+            m = maplikeOrSetlikeOrIterable
```
Comment (jdm) [inline components/script/dom/bindings/codegen/CodegenRust.py]: What is the purpose of this change?

**Comment by jdm** [comment]
Not super confidence-boosting that fixing the scoping issue triggered no test result changes in idlharness tests. Oh well!
@bors-servo r+

**Comment by nox** [comment]
> Not super confidence-boosting that fixing the scoping issue triggered no test result changes in idlharness tests. Oh well!

AFAIK this block of code only uses object properties that are common to all descriptors of the interface for which we are generating stuff. To make it an actual bug, you need to do some different `[Pref]` attributes on different interface members and whatnot, with the last one getting different gates than the ones you want for the iterator symbols.

**Comment by jdm** [comment]
@bors-servo retry
* #25513
